scipy/integrate/__quadpack.h

```diff
@@ -521,23 +567,23 @@ static PyObject *quadpack_qagpe(PyObject *dummy, PyObject *args) {
   points = (double *)ap_points->data;
 
   /* Set up iwork and work arrays */
-  ap_iord = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,NPY_INT);
+  ap_iord = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,F_INT_NPY);
   ap_alist = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,NPY_DOUBLE);
   ap_blist = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,NPY_DOUBLE);
   ap_rlist = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,NPY_DOUBLE);
   ap_elist = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,NPY_DOUBLE);
   ap_pts = (PyArrayObject *)PyArray_SimpleNew(1,npts2_shape,NPY_DOUBLE);
-  ap_level = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,NPY_DOUBLE);
-  ap_ndin = (PyArrayObject *)PyArray_SimpleNew(1,npts2_shape,NPY_DOUBLE);
+  ap_level = (PyArrayObject *)PyArray_SimpleNew(1,limit_shape,F_INT_NPY);
+  ap_ndin = (PyArrayObject *)PyArray_SimpleNew(1,npts2_shape,F_INT_NPY);
   if (ap_iord == NULL || ap_alist == NULL || ap_blist == NULL || ap_rlist == NULL || ap_elist == NULL || ap_pts == NULL || ap_level == NULL || ap_ndin == NULL) goto fail;
-  iord = (int *)ap_iord->data;
+  iord = (F_INT *)ap_iord->data;
   alist = (double *)ap_alist->data;
   blist = (double *)ap_blist->data;
   rlist = (double *)ap_rlist->data;
   elist = (double *)ap_elist->data;
   pts = (double *)ap_pts->data;
-  level = (int *)ap_level->data;
-  ndin = (int *)ap_level->data;
+  level = (F_INT *)ap_level->data;
+  ndin = (F_INT *)ap_ndin->data;
```

Comment:
Note that the code in the old version seemed to have a bug here --- moreover, ap_level/ap_ndin arrays had the wrong types.